docs/arch/adr-010-new-versioning-schema.md

```diff
@@ -0,0 +1,119 @@
+# ADR 10: kURL new versioning scheme
+
+## Context
+Based on our prior experience, offering customers the ability to select various add-on versions provides them with significant flexibility, but it also creates numerous challenges. Presently, we're struggling to ensure that each customer's chosen add-ons are compatible with one another and that migration paths are available for all potential combinations. Moreover our user feedback indicates that stability and predictability are their top priorities, rather than the specific technologies they are running behind the scenes.
+
+## Decision
+Based on users feedback and the struggling we are facing to maintain kURL upgrades predictability, we have decided to adopt a more opinionated approach to delivering our Kubernetes distribution. This will involve implementing a well-designed versioning scheme, which is outlined in detail in this Architecture Decision Record.
+
+## Solution
+Under the new versioning scheme, our offering will be organised into Channels that correspond to specific Kubernetes minor versions. As of now, we will begin with a single channel named v1.27, which serves as our official offering for Kubernetes versions v1.27.x. Put simply, users who wish to install or keep their installations on Kubernetes v1.27 will remain on this Channel.
+
+Consequently, we can initiate the design of our new kURL installation YAML in the following manner:
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27”
+```
+
+With the Channel (Kubernetes) now established as the basis for our delivery model, we can enable users to select a Version within this channel, which will allow them to access a specific set of Kubernetes-related features and functionalities (add-ons) that have been thoroughly tested and vetted by our team. This Version serves to abstract each individual add-on version into our own standardised version, ensuring that users can reliably and confidently utilise our offering without worrying about compatibility or interoperability issues.
+
+We will establish the practice of starting each Channel with a version v1.0.0 that groups together the initial set of add-ons (and respective versions) we support for that particular version. As a result, we will need to update the YAML file to reflect this approach, an initial Installer would look like this:
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27"
+  version:  "1.0.0"
```

Comment:
im trying to think about how much sense it makes to tie these two versions in the same spec. the way i see it the version determines the binary installer version or the scripts version that is downloaded and then the channel or any other config will be used as configuration to the install command. it doesnt make a whole lot of sense to add the "version" to the latter. i dont have an answer for you only something to think about. for example:

```
curl https://get.kurl.sh | sudo VERSION=1.0.1 sh # this downloads binary version 1.0.1
kurl install --config=kurl-config.yaml # specifies the k8s version to use and the add-on config
```

Reply:
I don't see why one would need to set an environment variable. The way I was looking at this was:
```shell
$ curl https://get.kurl.sh/<sha>
```
The service answering this request would discover what Version from what Channel to provide the installer based on reading the `<sha>`.  Assuming that `<sha>` holds the following Installer:
```yaml
apiVersion: "cluster.kurl.sh/v1"
kind: "Installer"
metadata:
  name: "cluster"
spec:
  channel:  "1.27"
  version:  "1.8.0"
```
The service, on this case, would render and answer the following script (just to ilustrate):
```shell
#!/bin/bash

installer="
apiVersion: cluster.kurl.sh/v1
kind: Installer
metadata:
  name: cluster
spec:
  channel:  1.27
  version:  1.8.0
"

echo $installer > config.yaml
curl -O https://get.kurl.sh/release/1.27/1.8.0/kurl.bin
./kurl.bin --config ./config.yaml
```

The Channel and Version parts of the URL path (see the curl command above) would be replaced by the service answering to the initial call (I used `kurl.bin` so it is easier to understand that we are downloading a binary). 

When running the installer (`./kurl.bin --config ./config.yaml`) both values for `channel` and `version` are obsolete but we still need to include the installer as it may also contain add-ons configuration.

Another possibility of installation: manually downloading the release binary and then running it. On this case both `version` and `channel` are useless as those were already chosen by the user when downloading the binary.

In essence, the Version and Channel are useful only when a given vendor wants to pin their Installer to an specific release (most likely in the vendor portal).